conda/logic.py

```diff
@@ -331,7 +331,74 @@ def LB_Preprocess_(self, equation):
         equation = sorted(equation)
         return equation, offset
 
-    def LinearBound_(self, equation, lo, hi, preprocess, polarity):
+    def Sorter_(self, equation, polarity=None):
+        explode = sum(([a] * c for c, a in equation), [])
+        nterms = len(explode)
+        explode.extend([False] * (2**ceil(log2(nterms)) - nterms))
+
+        def cmp(a, b):
+            aa, bb = explode[a], explode[b]
+            explode[a], explode[b] = self.Or(aa, bb, polarity), self.And(aa, bb, polarity)
+
+        def merge(lo, hi, r):
+            step = r * 2
+            if step < hi - lo:
+                merge(lo, hi, step)
+                merge(lo + r, hi, step)
+                for i in range(lo + r, hi - r, step):
+                    cmp(i, i + r)
+            else:
+                cmp(lo, lo + r)
+
+        def sort(lo, hi):
+            if hi - lo >= 1:
+                mid = lo + ((hi - lo) // 2)
+                sort(lo, mid)
+                sort(mid + 1, hi)
+                merge(lo, hi, 1)
+
+        sort(0, len(explode) - 1)
+        return explode
+
+    def BDD_(self, equation, nterms, lo, hi, polarity):
+        # The equation is sorted in order of increasing coefficients.
```

Comment:
Thank you for starting to break apart some of these HUGE functions and methods :)